dvc/ignore.py

```diff
@@ -222,7 +240,8 @@ def _update(self, dirname):
     def _update_sub_repo(self, root, dirs):
         for d in dirs:
             if self._is_dvc_repo(root, d):
-                new_pattern = DvcIgnorePatterns(["/{}/".format(d)], root)
+                self._ignored_subrepos.add(f"{root}{os.sep}{d}")
```

Comment:
Why not `os.path.join` ? If for perf reasons, then it is probably not a concern here, as we do this once for every parent in a path, which is negligible.